pheno_images/curate_weather_forecasts.R

```diff
@@ -1,7 +1,44 @@
+###########Read in libraries###########
 # devtools::install_github('eco4cast/neon4cast')
 # install.packages('readr')
-library(neon4cast)
-pheno_sites <- c("HARV", "BART", "SCBI", "STEI", "UKFS", "GRSM", "DELA", "CLBJ)"
-download_noaa(pheno_sites)
+# install.packages('tidync')
+# library(neon4cast)
+library(readr)
+library(dplyr)
+library(udunits2)
+library(plantecophys)
+source("~/neon4cast/R/noaa.R")
+
+###########Download weather data###########
+pheno_sites <- c("HARV", "BART", "SCBI", "STEI", "UKFS", "GRSM", "DELA", "CLBJ")
+download_noaa(siteID = pheno_sites, interval = "1hr", date = Sys.Date() - 1)
 noaa_fc <- stack_noaa()
-readr::write_csv(noaa_fc, file = paste0('pheno_images/NOAA_GEFS_35d_', Sys.Date(), '.csv'))
+
+###########Clean up weather data###########
+# First, convert units and take median of 31 ensembles
+hourly <- noaa_fc %>% 
+  tidyr::drop_na() %>% # the 36th day has NAs, exclude
+  mutate(airtemp_C = ud.convert(air_temperature, "kelvin", "celsius"),
+         precip = ud.convert(precipitation_flux, "s^-1", "d^-1"), #kg per m2 is equivalent to mm
+         vpd = RHtoVPD(RH = relative_humidity, TdegC = airtemp_C)) %>%
+  group_by(siteID, time) %>%
+  summarize(radiation = median(surface_downwelling_shortwave_flux_in_air),
+            airtemp_C = median(airtemp_C),
+            precip = median(precip),
+            vpd = median(vpd),
+            rad_Mj_hr = ud.convert(radiation*60*60, "joule", "megajoule")) %>%
+  ungroup() %>%
+  mutate(date = as.Date(substr(time, 1, 19)))
```

Comment:
There's something going on with this conversion to date, I'm not sure what? The `time` column at that point is incrementing values starting at 0 per hour.

Reply:
I'm not sure what the issue is. Just ran it again and didn't get an error. Could you please clarify?